File: mes-dashboard/src/main/java/com/mes/dashboard/entity/DeviceStatus.java
package com.mes.dashboard.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.mes.common.entity.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.LocalDateTime;

/**
 * 设备状态实体类
 * @author MES
 * @description 存储设备状态信息，包括设备编码、名称、状态、心跳等
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("dash_device_status")
public class DeviceStatus extends BaseEntity {

    /** 设备编码 */
    private String deviceCode;

    /** 设备名称 */
    private String deviceName;

    /** 设备类型 */
    private String deviceType;

    /** 状态: ONLINE/OFFLINE/ALARM */
    private String status;

    /** 温度 */
    private Double temperature;

    /** 速度 */
    private Double speed;

    /** 最后心跳时间 */
    private LocalDateTime lastHeartbeat;

    /** 工位ID */
    private Long workstationId;

    /** 生产线ID */
    private Long productionLineId;
}

File: mes-device-simulator-wpf/DeviceStatus.cs
using Newtonsoft.Json;

namespace MESDeviceSimulator;

public class DeviceStatus
{
    [JsonProperty("id")]
    public long Id { get; set; }

    [JsonProperty("deviceCode")]
    public string DeviceCode { get; set; } = "";

    [JsonProperty("deviceName")]
    public string DeviceName { get; set; } = "";

    [JsonProperty("deviceType")]
    public string DeviceType { get; set; } = "";

    [JsonProperty("status")]
    public string Status { get; set; } = "";

    [JsonProperty("temperature")]
    public double Temperature { get; set; }

    [JsonProperty("speed")]
    public double Speed { get; set; }
}